biodatasets/cord_ner/cord_ner.py

```diff
@@ -0,0 +1,333 @@
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        if self.config.data_dir is None:
+            # The download method may not be reliable, so if it fails this function will still work for local files
+            try:
+
+                if (
+                    self.config.name == "cord_ner_full_source"
+                    or self.config.name == "cord_ner_bigbio_kb"
+                ):
+                    url = {"train": _URLS[_DATASETNAME]["full"]}
+                elif self.config.name == "cord_ner_ner_source":
+                    url = {"train": _URLS[_DATASETNAME]["ner"]}
+                elif self.config.name == "cord_ner_corpus_source":
+                    url = {"train": _URLS[_DATASETNAME]["corpus"]}
+                data_dir = dl_manager.download_and_extract(url)
+            except:
+                raise ConnectionError(
+                    "The dataset could not be downloaded. Please download to local storage and pass the data_dir kwarg to load_dataset."
+                )
+        else:
+            filenames = {
+                "full": "CORD-NER-full.json",
+                "ner": "CORD-NER-ner.json",
+                "corpus": "CORD-NER-corpus.json",
+            }
+
+            if (
+                self.config.name == "cord_ner_full_source"
+                or self.config.name == "cord_ner_bigbio_kb"
+            ):
+                data_dir = {
+                    "train": os.path.join(self.config.data_dir, filenames["full"])
+                }
+            elif self.config.name == "cord_ner_ner_source":
+                data_dir = {
+                    "train": os.path.join(self.config.data_dir, filenames["ner"])
+                }
+            elif self.config.name == "cord_ner_corpus_source":
+                data_dir = {
+                    "train": os.path.join(self.config.data_dir, filenames["corpus"])
+                }
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_dir["train"],
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        if self.config.schema == "source":
+            with open(filepath) as fp:
+                for key, example in enumerate(fp.readlines()):
+                    yield key, json.loads(example)
+
+        elif self.config.schema == "bigbio_kb":
+
+            with open(filepath) as fp:
+                unq_id = 0
+                for key, line in enumerate(fp.readlines()):
+                    # There are some really long lines that can cause the build to crash.
+                    if len(line) > 1e5:
```

Comment:
I had to put this in otherwise my computer would crash. I'm curious to know if other people's computers will crash too. About 13% of the lines in the file have more than 100k characters, so this does ignore a sizable portion.

100k characters is kind of an absurd length for NER. Should I break it up into multiple examples?